citation-server/src/graphql/mutation.js

```diff
@@ -39,80 +40,145 @@ export const LinksInputType = new GraphQLInputObjectType({
 	})
 });
 
-export const PageInputType = new GraphQLInputObjectType({
-	name: 'PageInput',
-	fields: () => ({
-		__id__: {type: GraphQLID},
-		__newId__: {type: GraphQLID},
-		slug: {type: GraphQLString},
-		title: {type: GraphQLString},
-		children: {type: LinksInputType},
-		component: {type: LinkInputType}
-	})
-});
-
-export const ComponentInputType = new GraphQLInputObjectType({
-	name: 'ComponentInput',
-	fields: () => ({
-		__id__: {type: GraphQLID},
-		__newId__: {type: GraphQLID},
-		type: {type: GraphQLString},
-		children: {type: LinksInputType},
-		data: {type: LinksInputType}
-	})
-});
-
-export const ContentInputType = new GraphQLInputObjectType({
-	name: 'ContentInput',
-	fields: () => ({
-		__id__: {type: GraphQLID},
-		__newId__: {type: GraphQLID},
-		title: {type: GraphQLString},
-		content: {type: GraphQLString}
-	})
-});
+// export const PageInputType = new GraphQLInputObjectType({
+// 	name: 'PageInput',
+// 	fields: () => ({
+// 		__id__: {type: GraphQLID},
+// 		__newId__: {type: GraphQLID},
+// 		slug: {type: GraphQLString},
+// 		title: {type: GraphQLString},
+// 		children: {type: LinksInputType},
+// 		component: {type: LinkInputType}
+// 	})
+// });
+//
+// export const ComponentInputType = new GraphQLInputObjectType({
+// 	name: 'ComponentInput',
+// 	fields: () => ({
+// 		__id__: {type: GraphQLID},
+// 		__newId__: {type: GraphQLID},
+// 		type: {type: GraphQLString},
+// 		children: {type: LinksInputType},
+// 		data: {type: LinksInputType}
+// 	})
+// });
+//
+// export const ContentInputType = new GraphQLInputObjectType({
+// 	name: 'ContentInput',
+// 	fields: () => ({
+// 		__id__: {type: GraphQLID},
+// 		__newId__: {type: GraphQLID},
+// 		title: {type: GraphQLString},
+// 		content: {type: GraphQLString}
+// 	})
+// });
 
-export default new GraphQLObjectType({
-	name: 'Mutation',
-	fields: {
-		editPage: {
-			type: PageType,
-			args: {page: {type: PageInputType}},
-			resolve: async (root, params) => {
-				const {page} = params;
-				logger.debug(`mutation ${page}`);
-				try {
-					return await writeObject('Page', page);
-				} catch (error) {
-					throw error;
-				}
-			}
-		},
-		editComponent: {
-			type: ComponentType,
-			args: {component: {type: ComponentInputType}},
-			resolve: async (root, params) => {
-				const {component} = params;
-				logger.debug(`mutation ${component}`);
-				try {
-					return await writeObject('Component', component);
-				} catch (error) {
-					throw error;
-				}
+async function buildInputs() {
+	const InputType = {};
+	const model = await readModel();
+	for (const structure of model) {
+		InputType[`${structure.name}`] = new GraphQLInputObjectType({
```

Comment:
``InputType[`${stringValue}`]`` is over-engineered, it's equivalent to ``InputType[stringValue]``  (same many times)